### ISSUE TITLE
Selectable Label/Icon component

### ISSUE THREAD

**Issue by nah0y** [InvexGames/MaterialUI#8]
This would be awesome, and I think we need that almost in every game/software: http://www.google.fr/design/spec/components/chips.html

It's basically just a "row" component, that act like a button, you put everything you want in the row, and it's clickable as a button.
Currently I can't take a common button and use it for that, because when I resize it, it breaks the background and shadow of the button.

**Comment by nah0y** [comment]
Okay, it seems I can just create a panel, and add a Button and RippleConfig script on it? Can you confirm this is the correct way to do that?
Even if it is, it would be great to have a "ButtonPanel" component that comes as a predefined prefab :)

**Comment by InvexGames** [comment]
I'm not quite sure exactly what it is that you mean. Did you want something that exactly replicates the chip's functionality, or something like the [SelectionBox](https://github.com/InvexGames/MaterialUI/wiki/SelectionBox)?

**Comment by nah0y** [comment]
I think this tool need something like a fully selectable "complex" component.
It's just a background (any size we want) that act like a button (hover, ripple etc...) and on top of it we can put whatever we want, icon on the left + text for example.

In the Material example, Chips can be dragged etc... but what I'm talking about is just a simple button (with material graphisme) on which we can put what we want.
So basically, it's just the Panel you already have, but with a Button attached and the Ripple script. And I think the default size should be something like 200x50 (not the big square that you have with the panel).

If you still don't see what I'm talking about, I can make a gif :p

**Comment by InvexGames** [comment]
Ohh, I think I know now. Gimme a sec.

**Comment by InvexGames** [comment]
You mean something like this (that you can resize)?

![example](http://i.minus.com/iVKPnhoheVavj.gif)

**Comment by nah0y** [comment]
Exactly :)
So we could use that for example in a scroll list of contacts.
So maybe without the shadow?

**Comment by InvexGames** [comment]
So more like this? (Sorry for the delay)

![example](http://i.imgur.com/rfwDx4z.gif)

**Comment by nah0y** [comment]
Hum... Yes exactly like this :)

**Comment by InvexGames** [comment]
Awesome. I was planning on making a list-type object anyway. I'll get working on it :)

**Comment by InvexGames** [comment]
Actually I'll leave it open till it's added.

**Comment by InvexGames** [comment]
They've been added now, however, many things are still missing, namely:
- Easy-to-manage dividers
- Scrollbar management
- Overscroll (getting it to look like lollipop might take a fair amount of work, but I want it so bad!)

**Comment by nah0y** [comment]
Ahahah really nice!
For the overscroll... yes, it would be awesome, but certainly not the most important thing to do :)

**Comment by InvexGames** [comment]
Also, I want to give RippleConfig an option to delay the ripple by a frame or two, if it's part of a list. That way if you drag the list straight away, it won't create a ripple.

**Comment by nah0y** [comment]
+100

**Comment by grosssmutttr** [comment]
Tested the current ListView/ListItems and the RippleEffect appears great on a pc but on mobile devices it is only displayed sometimes...perhaps this is because of the Mesh.createVBO performance issue.

**Comment by InvexGames** [comment]
Hi! Thanks for bringing this issue to my attention :)

I'm now working on MaterialUI in a private repo, adding features and the likes for an official release on the Asset Store - this means that I won't be adding/fixing anything more in this repo (sorry). Feel free to fork this one if you'd like to tackle to issue yourself, and/or send me an email at invex@invexgames.com if you want me to let you know when it's available.

Cheers,
Declan.
